smac/facade/abstract_facade.py

```diff
@@ -275,7 +275,7 @@ def tell(self, info: TrialInfo, value: TrialValue, save: bool = True) -> None:
         """
         return self._optimizer.tell(info, value, save=save)
 
-    def optimize(self) -> Configuration | list[Configuration]:
+    def optimize(self, shared_data=None) -> Configuration | list[Configuration]:
```

Comment:
Can we add a proper type to it? I'm not sure if it's of a special type or just a `dict[str, Any]`? A brief comment about the argument would also be helpful :)

```suggestion
    def optimize(self, shared_data: dict[str, Any] | None = None) -> Configuration | list[Configuration]:
```